src/api/app/controllers/webui/staging/workflows_controller.rb

```diff
@@ -89,17 +89,21 @@ def update
       flash[:error] = "Managers group couldn't be assigned: #{@staging_workflow.errors.full_messages.to_sentence}."
     end
 
-    redirect_to edit_staging_workflow_path(@staging_workflow)
+    redirect_to edit_staging_workflow_path(@staging_workflow.project)
   end
 
   private
 
+  def set_workflow_project
+    @project = Project.find_by!(name: params[:workflow_project])
+  end
+
   def set_staging_workflow
-    @staging_workflow = ::Staging::Workflow.find_by(id: params[:id])
+    @staging_workflow = @project.staging
     return if @staging_workflow
 
     redirect_back(fallback_location: root_path)
-    flash[:error] = "Staging with id = #{params[:id]} doesn't exist"
+    flash[:error] = 'Staging project not found'
```

Comment:
This message could be confusing, because what we don't found is the `Staging` and not the `Staging project`, maybe something like ``` "Project #{@project} doesn't have a Staging"```

Reply:
done